packages/calypso-e2e/src/lib/pages/plans-page.ts

```diff
@@ -0,0 +1,87 @@
+import { Page } from 'playwright';
+import { getViewportName } from '../../browser-helper';
+
+// Types to restrict the string arguments passed in. These are fixed sets of strings, so we can be more restrictive.
+export type Plan = 'Free' | 'Personal' | 'Premium' | 'Business' | 'eCommerce';
+export type PlansPageTab = 'My Plan' | 'Plans';
+export type PlanActionButton = 'Manage plan' | 'Upgrade';
+
+const dynamicSelectors = {
+	myPlanTitle: ( planName: Plan ) => `.my-plan-card__title:has-text("${ planName }")`,
+	navigationTab: ( tabName: PlansPageTab ) => `.section-nav-tab:has-text("${ tabName }")`,
+	activeNavigationTab: ( tabName: PlansPageTab ) =>
+		`.is-selected.section-nav-tab:has-text("${ tabName }")`,
+	mobileActionButton: ( { plan, buttonText }: { plan: Plan; buttonText: PlanActionButton } ) =>
+		`.plan-features__mobile >> .plan-features__actions-button.is-${ plan.toLowerCase() }-plan:has-text("${ buttonText }")`,
+	desktopActionButton: ( { plan, buttonText }: { plan: Plan; buttonText: PlanActionButton } ) =>
+		`.plan-features__table >> .plan-features__actions-button.is-${ plan.toLowerCase() }-plan:has-text("${ buttonText }")`,
```

Comment:
Just a thought - since the desktop and mobile version only differ by the first bit (`.plan-features__<viewport>`, would this selector work instead?

```
.plan-features__${viewport} >> .plan-features__actions-button.is-${ plan.toLowerCase() }-plan:has-text("${ buttonText }")
```

By unifying the selector it should be possible to call the same selector, just with an additional param for viewport:

```
const viewportType = getViewportName();
const selector = dynamicSelectors.mobileActionButton( { viewport: viewport, plan: plan, buttonText: buttonText } )
```

If you have already tried unifying the two selectors and it didn't work, then sorry about that!